// react-native/Libraries/NewAppScreen/components/LearnMoreLinks.js
/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

import type {Node} from 'react';

import TouchableOpacity from '../../Components/Touchable/TouchableOpacity';
import View from '../../Components/View/View';
import openURLInBrowser from '../../Core/Devtools/openURLInBrowser';
import StyleSheet from '../../StyleSheet/StyleSheet';
import Text from '../../Text/Text';
import useColorScheme from '../../Utilities/useColorScheme';
import Colors from './Colors';
import React, {Fragment} from 'react';
import {Linking} from 'react-native';

const links = [
  {
    id: 1,
    title: 'Instagram',
    link: 'https://instagram.com/reioreoo',
    description: 'https://instagram.com/reioreoo',
  },
  {
    id: 2,
    title: 'LinkdIn',
    link: 'https://www.linkedin.com/in/anissasephia',
    description:
      'https://www.linkedin.com/in/anissasephia',
  },
  {
    id: 3,
    title: 'Github',
    link: 'https://github.com/anissasephia',
    description: 'https://github.com/anissasephia',
  },
  {
    id: 4,
    title: 'Discord',
    link: 'https://discordapp.com/users/#ansse',
    description: '#ansse',
  },
  {
    id: 5,
    title: 'Email',
    link: 'anissasephia@gmail.com',
    description:
      'anissasephia@gmail.com',
  },
  {
    id: 6,
    title: 'Address',
    //karene bersifat press oopenURLInBrowser di laptop jd butuh Linking.openURL pada onPress Di Line 95 agar dapat membuka di emulatornya
    link: 'google.navigation:q=-7.757867703072085, 110.37723381351542',
    
    description: 'Jalan pogung baru blok F32 Rt 15 Rw 052, Pogung Lor, Sinduadi, Kec. Mlati, Kabupaten Sleman, Daerah Istimewa Yogyakarta 55284',
  },
 
];

const LinkList = (): Node => {
  const isDarkMode = useColorScheme() === 'dark';
  return (
    <View style={styles.container}>
      {links.map(({id, title, link, description}) => (
        <Fragment key={id}>
          <View
            style={[
              styles.separator,
              {
                backgroundColor: isDarkMode ? Colors.dark : Colors.light,
              },
            ]}
          />
          <TouchableOpacity
            accessibilityRole="button"
            onPress={() => Linking.openURL(link)}
            style={styles.linkContainer}>
            <Text style={styles.link}>{title}</Text>
            <Text
              style={[
                styles.description,
                {
                  color: isDarkMode ? Colors.lighter : Colors.dark,
                },
              ]}>
              {description}
            </Text>
          </TouchableOpacity>
        </Fragment>
      ))}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginTop: 32,
    paddingHorizontal: 24,
  },
  linkContainer: {
    flexWrap: 'wrap',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingVertical: 8,
  },
  link: {
    flex: 2,
    fontSize: 18,
    fontWeight: '400',
    color: Colors.primary,
  },
  description: {
    flex: 3,
    paddingVertical: 16,
    fontWeight: '400',
    fontSize: 18,
  },
  separator: {
    height: StyleSheet.hairlineWidth,
  },
});

export default LinkList;
